app/src/editor/shared/components/StreamSelector.jsx

```diff
@@ -17,44 +18,60 @@ type Props = {
 }
 
 type State = {
+    loadedStream: any,
     isOpen: boolean,
     search: string,
     matchingStreams: Array<Object>,
 }
 
 export default class StreamSelector extends React.Component<Props, State> {
+    currentSearch: string
+
     state = {
+        loadedStream: undefined,
         isOpen: false,
         search: '',
         matchingStreams: [],
     }
 
     unmounted = false
 
-    async componentDidMount() {
-        const stream = await getStream(this.props.value)
+    componentDidMount() {
+        this.loadStream()
+    }
 
-        if (this.unmounted) { return }
+    componentWillUnmount() {
+        this.unmounted = true
+    }
+
+    componentDidUpdate(prevProps: Props) {
+        if (prevProps.value !== this.props.value) {
+            this.loadStream()
+        }
+    }
+
+    loadStream = async () => {
+        const { value } = this.props
+        const { loadedStream } = this.state
+        // do nothing if stream already loaded
+        if (loadedStream === value) { return }
+
+        const stream = await getStream(value)
+
+        if (this.unmounted || this.props.value !== value) { return }
```

Comment:
This line is kinda confusing, haha. The above `await` changes everything! ;)

Reply:
yeah really easy to make a mistakes here as well